<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Build stitched SpatialExperiment — build_SpatialExperiment • visiumStitched</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Build stitched SpatialExperiment — build_SpatialExperiment"><meta name="description" content='First, read in capture-area-level SpaceRanger
https://www.10xgenomics.com/support/software/space-ranger/latest/analysis/running-pipelines/space-ranger-count
outputs. Then, overwrite
spatial coordinates and images to represent group-level samples using
sample_info$group (though keep original coordinates in
colData columns ending with the suffix "_original"). Next, add info
about overlaps (via spe$exclude_overlapping and
spe$overlap_key).
Ultimately, return a
SpatialExperiment-class ready
for visualization or downstream analysis.'><meta property="og:description" content='First, read in capture-area-level SpaceRanger
https://www.10xgenomics.com/support/software/space-ranger/latest/analysis/running-pipelines/space-ranger-count
outputs. Then, overwrite
spatial coordinates and images to represent group-level samples using
sample_info$group (though keep original coordinates in
colData columns ending with the suffix "_original"). Next, add info
about overlaps (via spe$exclude_overlapping and
spe$overlap_key).
Ultimately, return a
SpatialExperiment-class ready
for visualization or downstream analysis.'><meta property="og:image" content="/logo.png"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">visiumStitched</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.12</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/visiumStitched.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/misc.html">Miscellaneous notes</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/LieberInstitute/visiumStitched/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Build stitched <code>SpatialExperiment</code></h1>
      <small class="dont-index">Source: <a href="https://github.com/LieberInstitute/visiumStitched/blob/devel/R/build_SpatialExperiment.R" class="external-link"><code>R/build_SpatialExperiment.R</code></a></small>
      <div class="d-none name"><code>build_SpatialExperiment.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>First, read in capture-area-level <code>SpaceRanger</code>
<a href="https://www.10xgenomics.com/support/software/space-ranger/latest/analysis/running-pipelines/space-ranger-count" class="external-link">https://www.10xgenomics.com/support/software/space-ranger/latest/analysis/running-pipelines/space-ranger-count</a>
outputs. Then, overwrite
spatial coordinates and images to represent group-level samples using
<code>sample_info$group</code> (though keep original coordinates in
<code>colData</code> columns ending with the suffix <code>"_original"</code>). Next, add info
about overlaps (via <code>spe$exclude_overlapping</code> and
<code>spe$overlap_key</code>).
Ultimately, return a
<a href="https://rdrr.io/pkg/SpatialExperiment/man/SpatialExperiment.html" class="external-link">SpatialExperiment-class</a> ready
for visualization or downstream analysis.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">build_SpatialExperiment</span><span class="op">(</span></span>
<span>  <span class="va">sample_info</span>,</span>
<span>  <span class="va">coords_dir</span>,</span>
<span>  count_type <span class="op">=</span> <span class="st">"sparse"</span>,</span>
<span>  reference_gtf <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  gtf_cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"source"</span>, <span class="st">"type"</span>, <span class="st">"gene_id"</span>, <span class="st">"gene_version"</span>, <span class="st">"gene_name"</span>, <span class="st">"gene_type"</span><span class="op">)</span>,</span>
<span>  calc_error_metrics <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-sample-info">sample_info<a class="anchor" aria-label="anchor" href="#arg-sample-info"></a></dt>
<dd><p>A <code><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame()</a></code> with columns <code>capture_area</code>,
<code>group</code>, <code>fiji_xml_path</code>, <code>fiji_image_path</code>,
<code>spaceranger_dir</code>, <code>intra_group_scalar</code>, and
<code>group_hires_scalef</code>. The last two are made by <code><a href="rescale_fiji_inputs.html">rescale_fiji_inputs()</a></code>.</p></dd>


<dt id="arg-coords-dir">coords_dir<a class="anchor" aria-label="anchor" href="#arg-coords-dir"></a></dt>
<dd><p>A <code>character(1)</code> vector giving the directory
containing sample directories each with <code>tissue_positions.csv</code>,
<code>scalefactors_json.json</code>, and <code>tissue_lowres_image.png</code> files
produced from refinement with <a href="prep_fiji.html">prep_fiji_coords()</a>
and related functions.</p></dd>


<dt id="arg-count-type">count_type<a class="anchor" aria-label="anchor" href="#arg-count-type"></a></dt>
<dd><p>A <code>character(1)</code> vector passed to <code>type</code> from
<code><a href="https://rdrr.io/pkg/SpatialExperiment/man/read10xVisium.html" class="external-link">SpatialExperiment::read10xVisium</a></code>, defaulting to "sparse".</p></dd>


<dt id="arg-reference-gtf">reference_gtf<a class="anchor" aria-label="anchor" href="#arg-reference-gtf"></a></dt>
<dd><p>Passed to <code><a href="https://rdrr.io/pkg/spatialLIBD/man/read10xVisiumWrapper.html" class="external-link">spatialLIBD::read10xVisiumWrapper()</a></code>. If
working on the same system where SpaceRanger was run, the GTF will be
automatically found; otherwise a <code>character(1)</code> path may be supplied,
pointing to a GTF file of gene annotation to populate <code>rowData()</code> with.</p></dd>


<dt id="arg-gtf-cols">gtf_cols<a class="anchor" aria-label="anchor" href="#arg-gtf-cols"></a></dt>
<dd><p>Passed to <code><a href="https://rdrr.io/pkg/spatialLIBD/man/read10xVisiumWrapper.html" class="external-link">spatialLIBD::read10xVisiumWrapper()</a></code>. Columns
in the reference GTF to extract and populate <code>rowData()</code>.</p></dd>


<dt id="arg-calc-error-metrics">calc_error_metrics<a class="anchor" aria-label="anchor" href="#arg-calc-error-metrics"></a></dt>
<dd><p>A <code>logical(1)</code> vector indicating whether to
calculate error metrics related to mapping spots to well-defined array
coordinates. If <code>TRUE</code>, adds <code>euclidean_error</code> and
<code>shared_neighbors</code> spot-level metrics to the <code>colData()</code>. The former
indicates distance in number of inter-spot distances to "move" a spot to the
new array position; the latter indicates the fraction of neighbors for the
associated capture area that are retained after mapping, which can be quite
time-consuming to compute.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A <a href="https://rdrr.io/pkg/SpatialExperiment/man/SpatialExperiment.html" class="external-link">SpatialExperiment-class</a>
object with one sample per group specified in <code>sample_info</code> using
transformed pixel and array coordinates (including  in the
<code>spatialCoords()</code>).</p>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Nicholas J. Eagles</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co">########################################################################</span></span></span>
<span class="r-in"><span><span class="co">#   Prepare sample_info</span></span></span>
<span class="r-in"><span><span class="co">########################################################################</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">sample_info</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/tibble.html" class="external-link">tibble</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    group <span class="op">=</span> <span class="st">"Br2719"</span>,</span></span>
<span class="r-in"><span>    capture_area <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"V13B23-283_A1"</span>, <span class="st">"V13B23-283_C1"</span>, <span class="st">"V13B23-283_D1"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="co">#   Add 'spaceranger_dir' column</span></span></span>
<span class="r-in"><span><span class="va">sr_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">temp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/unzip.html" class="external-link">unzip</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    <span class="fu">spatialLIBD</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/spatialLIBD/man/fetch_data.html" class="external-link">fetch_data</a></span><span class="op">(</span><span class="st">"visiumStitched_brain_spaceranger"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    exdir <span class="op">=</span> <span class="va">sr_dir</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2024-10-24 20:15:35.672125 loading file /github/home/.cache/R/BiocFileCache/7603a007f35_visiumStitched_brain_spaceranger.zip%3Frlkey%3Dbdgjc6mgy1ierdad6h6v5g29c%26dl%3D1</span>
<span class="r-in"><span><span class="va">sample_info</span><span class="op">$</span><span class="va">spaceranger_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    <span class="va">sr_dir</span>, <span class="va">sample_info</span><span class="op">$</span><span class="va">capture_area</span>, <span class="st">"outs"</span>, <span class="st">"spatial"</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">#   Add Fiji-output-related columns</span></span></span>
<span class="r-in"><span><span class="va">fiji_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">temp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/unzip.html" class="external-link">unzip</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    <span class="fu">spatialLIBD</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/spatialLIBD/man/fetch_data.html" class="external-link">fetch_data</a></span><span class="op">(</span><span class="st">"visiumStitched_brain_Fiji_out"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    exdir <span class="op">=</span> <span class="va">fiji_dir</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2024-10-24 20:15:38.221787 loading file /github/home/.cache/R/BiocFileCache/76048672d76_visiumStitched_brain_fiji_out.zip%3Frlkey%3Dptwal8f5zxakzejwd0oqw0lhj%26dl%3D1</span>
<span class="r-in"><span><span class="va">sample_info</span><span class="op">$</span><span class="va">fiji_xml_path</span> <span class="op">&lt;-</span> <span class="va">temp</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"xml$"</span>, <span class="va">temp</span><span class="op">)</span><span class="op">]</span></span></span>
<span class="r-in"><span><span class="va">sample_info</span><span class="op">$</span><span class="va">fiji_image_path</span> <span class="op">&lt;-</span> <span class="va">temp</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"png$"</span>, <span class="va">temp</span><span class="op">)</span><span class="op">]</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Re-size images and add more information to the sample_info</span></span></span>
<span class="r-in"><span><span class="va">sample_info</span> <span class="op">&lt;-</span> <span class="fu"><a href="rescale_fiji_inputs.html">rescale_fiji_inputs</a></span><span class="op">(</span><span class="va">sample_info</span>, out_dir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Preparing Fiji coordinates and images for build_SpatialExperiment()</span></span></span>
<span class="r-in"><span><span class="va">spe_input_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="prep_fiji.html">prep_fiji_coords</a></span><span class="op">(</span><span class="va">sample_info</span>, out_dir <span class="op">=</span> <span class="va">spe_input_dir</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "/tmp/RtmpphIhEc/Br2719/tissue_positions.csv"</span>
<span class="r-in"><span><span class="fu"><a href="prep_fiji.html">prep_fiji_image</a></span><span class="op">(</span><span class="va">sample_info</span>, out_dir <span class="op">=</span> <span class="va">spe_input_dir</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "/tmp/RtmpphIhEc/Br2719/tissue_lowres_image.png"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [2] "/tmp/RtmpphIhEc/Br2719/scalefactors_json.json" </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">########################################################################</span></span></span>
<span class="r-in"><span><span class="co">#   Build the SpatialExperiment</span></span></span>
<span class="r-in"><span><span class="co">########################################################################</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">#    Since we don't have access to the original GTF used to run SpaceRanger,</span></span></span>
<span class="r-in"><span><span class="co">#    we must explicitly supply our own GTF to build_SpatialExperiment(). We use</span></span></span>
<span class="r-in"><span><span class="co">#    GENCODE release 32, intended to be quite close to the actual GTF used,</span></span></span>
<span class="r-in"><span><span class="co">#    which is available from:</span></span></span>
<span class="r-in"><span><span class="co">#    https://cf.10xgenomics.com/supp/cell-exp/refdata-gex-GRCh38-2024-A.tar.gz</span></span></span>
<span class="r-in"><span><span class="va">bfc</span> <span class="op">&lt;-</span> <span class="fu">BiocFileCache</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocFileCache/man/BiocFileCache-class.html" class="external-link">BiocFileCache</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">gtf_cache</span> <span class="op">&lt;-</span> <span class="fu">BiocFileCache</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocFileCache/man/BiocFileCache-class.html" class="external-link">bfcrpath</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    <span class="va">bfc</span>,</span></span>
<span class="r-in"><span>    <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>        <span class="st">"ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/"</span>,</span></span>
<span class="r-in"><span>        <span class="st">"release_32/gencode.v32.annotation.gtf.gz"</span></span></span>
<span class="r-in"><span>    <span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Now we can build the stitched SpatialExperiment object</span></span></span>
<span class="r-in"><span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu">build_SpatialExperiment</span><span class="op">(</span></span></span>
<span class="r-in"><span>    <span class="va">sample_info</span>,</span></span>
<span class="r-in"><span>    coords_dir <span class="op">=</span> <span class="va">spe_input_dir</span>, reference_gtf <span class="op">=</span> <span class="va">gtf_cache</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Building SpatialExperiment using capture area as sample ID</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2024-10-24 20:15:47.126858 SpatialExperiment::read10xVisium: reading basic data from SpaceRanger</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2024-10-24 20:15:53.129575 read10xVisiumAnalysis: reading analysis output from SpaceRanger</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2024-10-24 20:15:53.448283 add10xVisiumAnalysis: adding analysis output from SpaceRanger</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2024-10-24 20:15:53.750296 rtracklayer::import: reading the reference GTF file</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2024-10-24 20:16:22.091238 adding gene information to the SPE object</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>Gene IDs did not match. This typically happens when you are not using the same GTF file as the one that was used by SpaceRanger. For example, one file uses GENCODE IDs and the other one ENSEMBL IDs. read10xVisiumWrapper() will try to convert them to ENSEMBL IDs.</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>Dropping 2226 out of 38606 genes for which we don't have information on the reference GTF file. This typically happens when you are not using the same GTF file as the one that was used by SpaceRanger.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2024-10-24 20:16:22.530429 adding information used by spatialLIBD</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overwriting imgData(spe) with merged images (one per group)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding array coordinates and overlap info</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Let's explore the stitched SpatialExperiment object</span></span></span>
<span class="r-in"><span><span class="va">spe</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> class: SpatialExperiment </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> dim: 36380 14976 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> metadata(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> assays(1): counts</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rownames(36380): ENSG00000243485.5 ENSG00000237613.2 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSG00000198695.2 ENSG00000198727.2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowData names(6): source type ... gene_type gene_search</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colnames(14976): AAACAACGAATAGTTC-1_V13B23-283_A1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   AAACAAGTATCTCCCA-1_V13B23-283_A1 ... TTGTTTGTATTACACG-1_V13B23-283_D1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   TTGTTTGTGTAAATTC-1_V13B23-283_D1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colData names(31): sample_id in_tissue ... overlap_key</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   exclude_overlapping</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> reducedDimNames(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> mainExpName: NULL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> altExpNames(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> spatialCoords names(2) : pxl_col_in_fullres pxl_row_in_fullres</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> imgData names(4): sample_id image_id data scaleFactor</span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Nicholas J. Eagles.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.9000.</p>
</div>

    </footer></div>





  </body></html>

