<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Build stitched SpatialExperiment — build_spe • visiumStitched</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Build stitched SpatialExperiment — build_spe"><meta property="og:description" content='First, read in capture-area-level SpaceRanger
https://www.10xgenomics.com/support/software/space-ranger/latest/analysis/running-pipelines/space-ranger-count
outputs. Then, overwrite
spatial coordinates and images to represent group-level samples using
sample_info$group (though keep original coordinates in
colData columns ending with the suffix "_original"). Next, add info
about overlaps (via spe$exclude_overlapping and
spe$overlap_key).
Ultimately, return a
SpatialExperiment-class ready
for visualization or downstream analysis.'><meta property="og:image" content="/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">visiumStitched</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.5</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/visiumStitched.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/misc.html">Miscellaneous notes</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/LieberInstitute/visiumStitched/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Build stitched <code>SpatialExperiment</code></h1>
    <small class="dont-index">Source: <a href="https://github.com/LieberInstitute/visiumStitched/blob/devel/R/build_spe.R" class="external-link"><code>R/build_spe.R</code></a></small>
    <div class="hidden name"><code>build_spe.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>First, read in capture-area-level <code>SpaceRanger</code>
<a href="https://www.10xgenomics.com/support/software/space-ranger/latest/analysis/running-pipelines/space-ranger-count" class="external-link">https://www.10xgenomics.com/support/software/space-ranger/latest/analysis/running-pipelines/space-ranger-count</a>
outputs. Then, overwrite
spatial coordinates and images to represent group-level samples using
<code>sample_info$group</code> (though keep original coordinates in
<code>colData</code> columns ending with the suffix <code>"_original"</code>). Next, add info
about overlaps (via <code>spe$exclude_overlapping</code> and
<code>spe$overlap_key</code>).
Ultimately, return a
<a href="https://rdrr.io/pkg/SpatialExperiment/man/SpatialExperiment.html" class="external-link">SpatialExperiment-class</a> ready
for visualization or downstream analysis.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">build_spe</span><span class="op">(</span></span>
<span>  <span class="va">sample_info</span>,</span>
<span>  <span class="va">coords_dir</span>,</span>
<span>  count_type <span class="op">=</span> <span class="st">"sparse"</span>,</span>
<span>  reference_gtf <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  gtf_cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"source"</span>, <span class="st">"type"</span>, <span class="st">"gene_id"</span>, <span class="st">"gene_version"</span>, <span class="st">"gene_name"</span>, <span class="st">"gene_type"</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-sample-info">sample_info<a class="anchor" aria-label="anchor" href="#arg-sample-info"></a></dt>
<dd><p>A <code><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame()</a></code> with columns <code>capture_area</code>,
<code>group</code>, <code>fiji_xml_path</code>, <code>fiji_image_path</code>,
<code>spaceranger_dir</code>, <code>intra_group_scalar</code>, and
<code>group_hires_scalef</code>. The last two are made by <code><a href="rescale_fiji_inputs.html">rescale_fiji_inputs()</a></code>.</p></dd>


<dt id="arg-coords-dir">coords_dir<a class="anchor" aria-label="anchor" href="#arg-coords-dir"></a></dt>
<dd><p>A <code>character(1)</code> vector giving the directory
containing sample directories each with <code>tissue_positions.csv</code>,
<code>scalefactors_json.json</code>, and <code>tissue_lowres_image.png</code> files
produced from refinement with <a href="prep_fiji_coords.html">prep_fiji_coords()</a>
and related functions.</p></dd>


<dt id="arg-count-type">count_type<a class="anchor" aria-label="anchor" href="#arg-count-type"></a></dt>
<dd><p>A <code>character(1)</code> vector passed to <code>type</code> from
<code><a href="https://rdrr.io/pkg/SpatialExperiment/man/read10xVisium.html" class="external-link">SpatialExperiment::read10xVisium</a></code>, defaulting to "sparse".</p></dd>


<dt id="arg-reference-gtf">reference_gtf<a class="anchor" aria-label="anchor" href="#arg-reference-gtf"></a></dt>
<dd><p>Passed to <code><a href="https://rdrr.io/pkg/spatialLIBD/man/read10xVisiumWrapper.html" class="external-link">spatialLIBD::read10xVisiumWrapper()</a></code>. If
working on the same system where SpaceRanger was run, the GTF will be
automatically found; otherwise a <code>character(1)</code> path may be supplied,
pointing to a GTF file of gene annotation to populate <code>rowData()</code> with.</p></dd>


<dt id="arg-gtf-cols">gtf_cols<a class="anchor" aria-label="anchor" href="#arg-gtf-cols"></a></dt>
<dd><p>Passed to <code><a href="https://rdrr.io/pkg/spatialLIBD/man/read10xVisiumWrapper.html" class="external-link">spatialLIBD::read10xVisiumWrapper()</a></code>. Columns
in the reference GTF to extract and populate <code>rowData()</code>.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>A <a href="https://rdrr.io/pkg/SpatialExperiment/man/SpatialExperiment.html" class="external-link">SpatialExperiment-class</a>
object with one sample per group specified in <code>sample_info</code> using
transformed pixel and array coordinates (including  in the
<code>spatialCoords()</code>).</p>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Nicholas J. Eagles</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co">########################################################################</span></span></span>
<span class="r-in"><span><span class="co">#   Prepare sample_info</span></span></span>
<span class="r-in"><span><span class="co">########################################################################</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.exists</a></span><span class="op">(</span><span class="st">"sample_info.rds"</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span>    <span class="va">sample_info</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">'sample_info.rds'</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span> <span class="kw">else</span> <span class="op">{</span></span></span>
<span class="r-in"><span>    <span class="va">sample_info</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/tibble.html" class="external-link">tibble</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>        group <span class="op">=</span> <span class="st">"Br2719"</span>,</span></span>
<span class="r-in"><span>        capture_area <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"V13B23-283_A1"</span>, <span class="st">"V13B23-283_C1"</span>, <span class="st">"V13B23-283_D1"</span><span class="op">)</span></span></span>
<span class="r-in"><span>    <span class="op">)</span></span></span>
<span class="r-in"><span>    <span class="co">#   Add 'spaceranger_dir' column</span></span></span>
<span class="r-in"><span>    <span class="va">sr_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span>    <span class="va">temp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/unzip.html" class="external-link">unzip</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>        <span class="fu">spatialLIBD</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/spatialLIBD/man/fetch_data.html" class="external-link">fetch_data</a></span><span class="op">(</span><span class="st">"visiumStitched_brain_spaceranger"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>        exdir <span class="op">=</span> <span class="va">sr_dir</span></span></span>
<span class="r-in"><span>    <span class="op">)</span></span></span>
<span class="r-in"><span>    <span class="va">sample_info</span><span class="op">$</span><span class="va">spaceranger_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>        <span class="va">sr_dir</span>, <span class="va">sample_info</span><span class="op">$</span><span class="va">capture_area</span>, <span class="st">"outs"</span>, <span class="st">"spatial"</span></span></span>
<span class="r-in"><span>    <span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span>    <span class="co">#   Add Fiji-output-related columns</span></span></span>
<span class="r-in"><span>    <span class="va">fiji_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span>    <span class="va">temp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/unzip.html" class="external-link">unzip</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>        <span class="fu">spatialLIBD</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/spatialLIBD/man/fetch_data.html" class="external-link">fetch_data</a></span><span class="op">(</span><span class="st">"visiumStitched_brain_Fiji_out"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>        exdir <span class="op">=</span> <span class="va">fiji_dir</span></span></span>
<span class="r-in"><span>    <span class="op">)</span></span></span>
<span class="r-in"><span>    <span class="va">sample_info</span><span class="op">$</span><span class="va">fiji_xml_path</span> <span class="op">&lt;-</span> <span class="va">temp</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"xml$"</span>, <span class="va">temp</span><span class="op">)</span><span class="op">]</span></span></span>
<span class="r-in"><span>    <span class="va">sample_info</span><span class="op">$</span><span class="va">fiji_image_path</span> <span class="op">&lt;-</span> <span class="va">temp</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"png$"</span>, <span class="va">temp</span><span class="op">)</span><span class="op">]</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span>    <span class="co">## Re-size images and add more information to the sample_info</span></span></span>
<span class="r-in"><span>    <span class="va">sample_info</span> <span class="op">&lt;-</span> <span class="fu"><a href="rescale_fiji_inputs.html">rescale_fiji_inputs</a></span><span class="op">(</span><span class="va">sample_info</span>, out_dir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span>    <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">saveRDS</a></span><span class="op">(</span><span class="va">sample_info</span>, <span class="st">"sample_info.rds"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Preparing Fiji coordinates and images for build_spe()</span></span></span>
<span class="r-in"><span><span class="va">spe_input_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="prep_fiji_coords.html">prep_fiji_coords</a></span><span class="op">(</span><span class="va">sample_info</span>, out_dir <span class="op">=</span> <span class="va">spe_input_dir</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "/tmp/RtmpfrwfBL/Br2719/tissue_positions.csv"</span>
<span class="r-in"><span><span class="fu"><a href="prep_fiji_image.html">prep_fiji_image</a></span><span class="op">(</span><span class="va">sample_info</span>, out_dir <span class="op">=</span> <span class="va">spe_input_dir</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "/tmp/RtmpfrwfBL/Br2719/tissue_lowres_image.png"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [2] "/tmp/RtmpfrwfBL/Br2719/scalefactors_json.json" </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">########################################################################</span></span></span>
<span class="r-in"><span><span class="co">#   Build the SpatialExperiment</span></span></span>
<span class="r-in"><span><span class="co">########################################################################</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">#    Since we don't have access to the original GTF used to run SpaceRanger,</span></span></span>
<span class="r-in"><span><span class="co">#    we must explicitly supply our own GTF to build_spe(). We use</span></span></span>
<span class="r-in"><span><span class="co">#    GENCODE release 32, intended to be quite close to the actual GTF used,</span></span></span>
<span class="r-in"><span><span class="co">#    which is available from:</span></span></span>
<span class="r-in"><span><span class="co">#    https://cf.10xgenomics.com/supp/cell-exp/refdata-gex-GRCh38-2024-A.tar.gz</span></span></span>
<span class="r-in"><span><span class="va">bfc</span> <span class="op">&lt;-</span> <span class="fu">BiocFileCache</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocFileCache/man/BiocFileCache-class.html" class="external-link">BiocFileCache</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">gtf_cache</span> <span class="op">&lt;-</span> <span class="fu">BiocFileCache</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocFileCache/man/BiocFileCache-class.html" class="external-link">bfcrpath</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    <span class="va">bfc</span>,</span></span>
<span class="r-in"><span>    <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>        <span class="st">"ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/"</span>,</span></span>
<span class="r-in"><span>        <span class="st">"release_32/gencode.v32.annotation.gtf.gz"</span></span></span>
<span class="r-in"><span>    <span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Now we can build the stitched SpatialExperiment object</span></span></span>
<span class="r-in"><span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu">build_spe</span><span class="op">(</span></span></span>
<span class="r-in"><span>    <span class="va">sample_info</span>,</span></span>
<span class="r-in"><span>    coords_dir <span class="op">=</span> <span class="va">spe_input_dir</span>, reference_gtf <span class="op">=</span> <span class="va">gtf_cache</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Building SpatialExperiment using capture area as sample ID</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2024-08-23 17:24:03.704573 SpatialExperiment::read10xVisium: reading basic data from SpaceRanger</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2024-08-23 17:24:10.618479 read10xVisiumAnalysis: reading analysis output from SpaceRanger</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2024-08-23 17:24:10.971693 add10xVisiumAnalysis: adding analysis output from SpaceRanger</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2024-08-23 17:24:11.300599 rtracklayer::import: reading the reference GTF file</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2024-08-23 17:24:41.005388 adding gene information to the SPE object</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>Gene IDs did not match. This typically happens when you are not using the same GTF file as the one that was used by SpaceRanger. For example, one file uses GENCODE IDs and the other one ENSEMBL IDs. read10xVisiumWrapper() will try to convert them to ENSEMBL IDs.</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>Dropping 2226 out of 38606 genes for which we don't have information on the reference GTF file. This typically happens when you are not using the same GTF file as the one that was used by SpaceRanger.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2024-08-23 17:24:41.240671 adding information used by spatialLIBD</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overwriting imgData(spe) with merged images (one per group)</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Let's explore the stitched SpatialExperiment object</span></span></span>
<span class="r-in"><span><span class="va">spe</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> class: SpatialExperiment </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> dim: 36380 14976 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> metadata(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> assays(1): counts</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rownames(36380): ENSG00000243485.5 ENSG00000237613.2 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSG00000198695.2 ENSG00000198727.2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowData names(6): source type ... gene_type gene_search</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colnames(14976): AAACAACGAATAGTTC-1_V13B23-283_A1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   AAACAAGTATCTCCCA-1_V13B23-283_A1 ... TTGTTTGTATTACACG-1_V13B23-283_D1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   TTGTTTGTGTAAATTC-1_V13B23-283_D1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colData names(31): sample_id in_tissue ... overlap_key</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   exclude_overlapping</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> reducedDimNames(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> mainExpName: NULL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> altExpNames(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> spatialCoords names(2) : pxl_col_in_fullres pxl_row_in_fullres</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> imgData names(4): sample_id image_id data scaleFactor</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Nicholas J. Eagles.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.9000.</p>
</div>

      </footer></div>






  </body></html>

