<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Introduction to visiumStitched • visiumStitched</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Introduction to visiumStitched">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">visiumStitched</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.8</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item"><a class="nav-link" href="../articles/visiumStitched.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/misc.html">Miscellaneous notes</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/LieberInstitute/visiumStitched/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Introduction to visiumStitched</h1>
                        <h4 data-toc-skip class="author">Nicholas J.
Eagles</h4>
            <address class="author_afil">
      Lieber Institute for Brain
Development<br><a class="author_email" href="mailto:#"></a><a href="mailto:nickeagles77@gmail.com" class="email">nickeagles77@gmail.com</a>
      </address>
                              <h4 data-toc-skip class="author">Leonardo
Collado-Torres</h4>
            <address class="author_afil">
      Lieber Institute for Brain DevelopmentCenter for Computational
Biology, Johns Hopkins UniversityDepartment of Biostatistics, Johns
Hopkins Bloomberg School of Public
Health<br><a class="author_email" href="mailto:#"></a><a href="mailto:lcolladotor@gmail.com" class="email">lcolladotor@gmail.com</a>
      </address>
                  
            <h4 data-toc-skip class="date">16 October 2024</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/LieberInstitute/visiumStitched/blob/devel/vignettes/visiumStitched.Rmd" class="external-link"><code>vignettes/visiumStitched.Rmd</code></a></small>
      <div class="d-none name"><code>visiumStitched.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="basics">Basics<a class="anchor" aria-label="anchor" href="#basics"></a>
</h2>
<div class="section level3">
<h3 id="install-visiumstitched">Install <code>visiumStitched</code><a class="anchor" aria-label="anchor" href="#install-visiumstitched"></a>
</h3>
<p><em><a href="https://bioconductor.org/packages/3.20/visiumStitched" class="external-link">visiumStitched</a></em>
is a Bioconductor <code>R</code> package that can be installed with the
following commands in your <code>R</code> session:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"LieberInstitute/visiumStitched"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="citing-visiumstitched">Citing <code>visiumStitched</code><a class="anchor" aria-label="anchor" href="#citing-visiumstitched"></a>
</h3>
<p>We hope that <em><a href="https://bioconductor.org/packages/3.20/visiumStitched" class="external-link">visiumStitched</a></em>
will be useful for your research. Please use the following information
to cite the package and the overall approach. Thank you!</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Citation info</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/citation.html" class="external-link">citation</a></span><span class="op">(</span><span class="st">"visiumStitched"</span><span class="op">)</span></span>
<span><span class="co">#&gt; To cite package 'visiumStitched' in publications use:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Eagles NJ, Collado-Torres L (2024). _Enable downstream analysis of</span></span>
<span><span class="co">#&gt;   Visium capture areas stitched together with Fiji_.</span></span>
<span><span class="co">#&gt;   doi:10.18129/B9.bioc.visiumStitched</span></span>
<span><span class="co">#&gt;   &lt;https://doi.org/10.18129/B9.bioc.visiumStitched&gt;,</span></span>
<span><span class="co">#&gt;   https://github.com/LieberInstitute/visiumStitched/visiumStitched - R</span></span>
<span><span class="co">#&gt;   package version 0.99.8,</span></span>
<span><span class="co">#&gt;   &lt;http://www.bioconductor.org/packages/visiumStitched&gt;.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Eagles NJ, Bach S, Tippani M, Ravichandran P, Du Y, Miller RA, Hyde</span></span>
<span><span class="co">#&gt;   TM, Page SC, Martinowich K, Collado-Torres L (2024).</span></span>
<span><span class="co">#&gt;   "visiumStitched." _bioRxiv_. doi:10.1101/TODO</span></span>
<span><span class="co">#&gt;   &lt;https://doi.org/10.1101/TODO&gt;,</span></span>
<span><span class="co">#&gt;   &lt;https://www.biorxiv.org/content/10.1101/TODO&gt;.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; To see these entries in BibTeX format, use 'print(&lt;citation&gt;,</span></span>
<span><span class="co">#&gt; bibtex=TRUE)', 'toBibtex(.)', or set</span></span>
<span><span class="co">#&gt; 'options(citation.bibtex.max=999)'.</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="packages-used-in-this-vignette">Packages used in this vignette<a class="anchor" aria-label="anchor" href="#packages-used-in-this-vignette"></a>
</h3>
<p>Let’s load the packages we’ll use in this vignette.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/drighelli/SpatialExperiment" class="external-link">"SpatialExperiment"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/LieberInstitute/visiumStitched" class="external-link">"visiumStitched"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://dplyr.tidyverse.org" class="external-link">"dplyr"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/LieberInstitute/spatialLIBD" class="external-link">"spatialLIBD"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"BiocFileCache"</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://ggplot2.tidyverse.org" class="external-link">"ggplot2"</a></span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="preparing-experiment-information">Preparing Experiment Information<a class="anchor" aria-label="anchor" href="#preparing-experiment-information"></a>
</h2>
<p>Much of the <code>visiumStitched</code> package uses a
<code>tibble</code> (or <code>data.frame</code>) defining information
about the experiment. Most fundamentally, the <code>group</code> column
allows you to line up which capture areas, in the
<code>capture_area</code> column, are to be stitched together later. In
our case, we have just one unique group, consisting of all three capture
areas. Note multiple groups are supported. By the end of this demo, the
<code>SpatialExperiment</code> will consist of just one sample composed
of the three capture areas; in general, there will be one sample per
group.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Create initial sample_info</span></span>
<span><span class="va">sample_info</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>    group <span class="op">=</span> <span class="st">"Br2719"</span>,</span>
<span>    capture_area <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"V13B23-283_A1"</span>, <span class="st">"V13B23-283_C1"</span>, <span class="st">"V13B23-283_D1"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Initial sample_info</span></span>
<span><span class="va">sample_info</span></span>
<span><span class="co">#&gt;    group  capture_area</span></span>
<span><span class="co">#&gt; 1 Br2719 V13B23-283_A1</span></span>
<span><span class="co">#&gt; 2 Br2719 V13B23-283_C1</span></span>
<span><span class="co">#&gt; 3 Br2719 V13B23-283_D1</span></span></code></pre></div>
<p>Next, we’ll need the Spaceranger outputs for each capture area, which
can be retrieved with <code><a href="https://rdrr.io/pkg/spatialLIBD/man/fetch_data.html" class="external-link">spatialLIBD::fetch_data()</a></code>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Download example SpaceRanger output files</span></span>
<span><span class="va">sr_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">temp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/unzip.html" class="external-link">unzip</a></span><span class="op">(</span><span class="fu">spatialLIBD</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/spatialLIBD/man/fetch_data.html" class="external-link">fetch_data</a></span><span class="op">(</span><span class="st">"visiumStitched_brain_spaceranger"</span><span class="op">)</span>,</span>
<span>    exdir <span class="op">=</span> <span class="va">sr_dir</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; 2024-10-16 19:27:29.854686 loading file /github/home/.cache/R/BiocFileCache/1c7541209b46_visiumStitched_brain_spaceranger.zip%3Frlkey%3Dbdgjc6mgy1ierdad6h6v5g29c%26dl%3D1</span></span>
<span><span class="va">sample_info</span><span class="op">$</span><span class="va">spaceranger_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span></span>
<span>    <span class="va">sr_dir</span>, <span class="va">sample_info</span><span class="op">$</span><span class="va">capture_area</span>, <span class="st">"outs"</span>, <span class="st">"spatial"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Sample_info with paths to SpaceRanger output directories</span></span>
<span><span class="va">sample_info</span></span>
<span><span class="co">#&gt;    group  capture_area                            spaceranger_dir</span></span>
<span><span class="co">#&gt; 1 Br2719 V13B23-283_A1 /tmp/RtmpyOAgfb/V13B23-283_A1/outs/spatial</span></span>
<span><span class="co">#&gt; 2 Br2719 V13B23-283_C1 /tmp/RtmpyOAgfb/V13B23-283_C1/outs/spatial</span></span>
<span><span class="co">#&gt; 3 Br2719 V13B23-283_D1 /tmp/RtmpyOAgfb/V13B23-283_D1/outs/spatial</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="preparing-inputs-to-fiji">Preparing Inputs to Fiji<a class="anchor" aria-label="anchor" href="#preparing-inputs-to-fiji"></a>
</h2>
<p>The <code>visiumStitched</code> workflow makes use of <a href="https://imagej.net/software/fiji/" class="external-link">Fiji</a>, a distribution of the
<code>ImageJ</code> image-processing software, which includes an
interface for aligning images on a shared coordinate system. Before
aligning anything in Fiji, we need to ensure that images to align from
all capture areas are on the same scale– that is, a pixel in each image
represents the same distance. This is typically approximately true, but
is not guaranteed to be exactly true, especially when the capture areas
to align come from different Visium slides.
<code><a href="../reference/rescale_fiji_inputs.html">rescale_fiji_inputs()</a></code> reads in the <a href="https://www.10xgenomics.com/support/software/space-ranger/latest/analysis/outputs/spatial-outputs#tissue-png" class="external-link">high-resolution
tissue images</a> for each capture area, and uses info about their spot
diameters in pixels and scale factors to rescale the images
appropriately (even if they are from different Visium slides).</p>
<p>For demonstration purposes, we’ll set <code>out_dir</code> to a
temporary location. Typically, it would really be any suitable directory
to place the rescaled images for later input to Fiji.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#   Generate rescaled approximately high-resolution images</span></span>
<span><span class="va">sample_info</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/rescale_fiji_inputs.html">rescale_fiji_inputs</a></span><span class="op">(</span><span class="va">sample_info</span>, out_dir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Sample_info with output directories</span></span>
<span><span class="va">sample_info</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 3 × 5</span></span></span>
<span><span class="co">#&gt;   group  capture_area  spaceranger_dir     intra_group_scalar group_hires_scalef</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                            <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>              <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> Br2719 V13B23-283_A1 /tmp/RtmpyOAgfb/V1…               1.00             0.082<span style="text-decoration: underline;">5</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> Br2719 V13B23-283_C1 /tmp/RtmpyOAgfb/V1…               1.00             0.082<span style="text-decoration: underline;">5</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> Br2719 V13B23-283_D1 /tmp/RtmpyOAgfb/V1…               1                0.082<span style="text-decoration: underline;">5</span></span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="building-a-spatialexperiment">Building a <code>SpatialExperiment</code><a class="anchor" aria-label="anchor" href="#building-a-spatialexperiment"></a>
</h2>
<div class="section level3">
<h3 id="stitching-images-with-fiji">Stitching Images with Fiji<a class="anchor" aria-label="anchor" href="#stitching-images-with-fiji"></a>
</h3>
<p>Before building a <code>SpatialExperiment</code> for a stitched
dataset, we must align the images for each group in Fiji. Check out <a href="https://www.youtube.com/watch?v=kFLtpK3qbSY" class="external-link">this video</a> for a
guide through this process with the example data.</p>
<iframe width="560" height="315" src="https://www.youtube.com/embed/kFLtpK3qbSY?si=CF-rv7Nd5yc7W6nC" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" referrerpolicy="strict-origin-when-cross-origin" allowfullscreen>
</iframe>
</div>
<div class="section level3">
<h3 id="creating-group-level-samples">Creating Group-Level Samples<a class="anchor" aria-label="anchor" href="#creating-group-level-samples"></a>
</h3>
<p>From the Fiji alignment, two output files will be produced: an
<code>XML</code> file specifying rigid affine transformations for each
capture area, and the stitched approximately high-resolution image.
These files for this dataset are available through
<code><a href="https://rdrr.io/pkg/spatialLIBD/man/fetch_data.html" class="external-link">spatialLIBD::fetch_data()</a></code>. We’ll need to add the paths to
the XML and PNG files to the <code>fiji_xml_path</code> and
<code>fiji_image_path</code> columns of <code>sample_info</code>,
respectively.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fiji_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">temp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/unzip.html" class="external-link">unzip</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/spatialLIBD/man/fetch_data.html" class="external-link">fetch_data</a></span><span class="op">(</span><span class="st">"visiumStitched_brain_Fiji_out"</span><span class="op">)</span>, exdir <span class="op">=</span> <span class="va">fiji_dir</span><span class="op">)</span></span>
<span><span class="co">#&gt; 2024-10-16 19:27:42.879193 loading file /github/home/.cache/R/BiocFileCache/1c754f81ded8_visiumStitched_brain_fiji_out.zip%3Frlkey%3Dptwal8f5zxakzejwd0oqw0lhj%26dl%3D1</span></span>
<span><span class="va">sample_info</span><span class="op">$</span><span class="va">fiji_xml_path</span> <span class="op">&lt;-</span> <span class="va">temp</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"xml$"</span>, <span class="va">temp</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">sample_info</span><span class="op">$</span><span class="va">fiji_image_path</span> <span class="op">&lt;-</span> <span class="va">temp</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"png$"</span>, <span class="va">temp</span><span class="op">)</span><span class="op">]</span></span></code></pre></div>
<p>We now have every column present in <code>sample_info</code> that
will be necessary for any <code>visiumStitched</code> function.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Complete sample_info</span></span>
<span><span class="va">sample_info</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 3 × 7</span></span></span>
<span><span class="co">#&gt;   group  capture_area  spaceranger_dir     intra_group_scalar group_hires_scalef</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                            <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>              <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> Br2719 V13B23-283_A1 /tmp/RtmpyOAgfb/V1…               1.00             0.082<span style="text-decoration: underline;">5</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> Br2719 V13B23-283_C1 /tmp/RtmpyOAgfb/V1…               1.00             0.082<span style="text-decoration: underline;">5</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> Br2719 V13B23-283_D1 /tmp/RtmpyOAgfb/V1…               1                0.082<span style="text-decoration: underline;">5</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 2 more variables: fiji_xml_path &lt;chr&gt;, fiji_image_path &lt;chr&gt;</span></span></span></code></pre></div>
<p>Before building the <code>SpatialExperiment</code>, the idea is to
create a directory structure very similar to <a href="https://www.10xgenomics.com/support/software/space-ranger/latest/analysis/outputs/spatial-outputs" class="external-link">Spaceranger’s
spatial outputs</a> for each <em>group</em>, as opposed to the
<em>capture-area</em>-level directories we already have. We’ll place
this directory in a temporary location that will later be read in to
produce the final <code>SpatialExperiment</code>.</p>
<p>First, <code><a href="../reference/prep_fiji.html">prep_fiji_coords()</a></code> will apply the rigid affine
transformations specified by Fiji’s output XML file to the spatial
coordinates, ultimately producing a group-level
<code>tissue_positions.csv</code> file. Next,
<code><a href="../reference/prep_fiji.html">prep_fiji_image()</a></code> will rescale the stitched image to have a
default of 1,200 pixels in the longest dimension. The idea is that in an
experiment with multiple groups, the images stored in the
<code>SpatialExperiment</code> for any group will be similarly scaled
and occupy similar memory footprints.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Prepare the Fiji coordinates and images.</span></span>
<span><span class="co">## These functions return the file paths to the newly-created files that follow</span></span>
<span><span class="co">## the standard directory structure from SpaceRanger (10x Genomics)</span></span>
<span><span class="va">spe_input_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/prep_fiji.html">prep_fiji_coords</a></span><span class="op">(</span><span class="va">sample_info</span>, out_dir <span class="op">=</span> <span class="va">spe_input_dir</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "/tmp/RtmpyOAgfb/Br2719/tissue_positions.csv"</span></span>
<span><span class="fu"><a href="../reference/prep_fiji.html">prep_fiji_image</a></span><span class="op">(</span><span class="va">sample_info</span>, out_dir <span class="op">=</span> <span class="va">spe_input_dir</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "/tmp/RtmpyOAgfb/Br2719/tissue_lowres_image.png"</span></span>
<span><span class="co">#&gt; [2] "/tmp/RtmpyOAgfb/Br2719/scalefactors_json.json"</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="constructing-the-object">Constructing the Object<a class="anchor" aria-label="anchor" href="#constructing-the-object"></a>
</h3>
<p>We now have all the pieces to create the
<code>SpatialExperiment</code> object. After constructing the base
object, information related to how spots may overlap between capture
areas in each <code>group</code> is added. The <code>sum_umi</code>
metric will by default determine which spots in overlapping regions to
exclude in plots. In particular, at regions of overlap, spots from
capture areas with higher average UMI (unique molecular identifier)
counts will be plotted, while any other spots will not be shown using
<code><a href="https://rdrr.io/pkg/spatialLIBD/man/vis_clus.html" class="external-link">spatialLIBD::vis_clus()</a></code>,
<code><a href="https://rdrr.io/pkg/spatialLIBD/man/vis_gene.html" class="external-link">spatialLIBD::vis_gene()</a></code>, and related visualization
functions. We’ll also mirror the image and gene-expression data to match
the orientation specified at the wet bench. More info about performing
geometric transformations is <a href="#geometric-transformations">here</a>.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Download the Gencode v32 GTF file which is the closest one to the one</span></span>
<span><span class="co">## that was used with SpaceRanger. Note that SpaceRanger GTFs are available at</span></span>
<span><span class="co">## https://cf.10xgenomics.com/supp/cell-exp/refdata-gex-GRCh38-2024-A.tar.gz</span></span>
<span><span class="co">## but is too large for us to download here since it includes many other files</span></span>
<span><span class="co">## we don't need right now.</span></span>
<span><span class="co">## However, ideally you would adapt this code and use the actual GTF file you</span></span>
<span><span class="co">## used when running SpaceRanger.</span></span>
<span><span class="va">bfc</span> <span class="op">&lt;-</span> <span class="fu">BiocFileCache</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocFileCache/man/BiocFileCache-class.html" class="external-link">BiocFileCache</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">gtf_cache</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocFileCache/man/BiocFileCache-class.html" class="external-link">bfcrpath</a></span><span class="op">(</span></span>
<span>    <span class="va">bfc</span>,</span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span></span>
<span>        <span class="st">"ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/"</span>,</span>
<span>        <span class="st">"release_32/gencode.v32.annotation.gtf.gz"</span></span>
<span>    <span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Now we can build the SpatialExperiment object. We'll later explore error</span></span>
<span><span class="co">## metrics related to computing new array coordinates, and thus specify</span></span>
<span><span class="co">## 'calc_error_metrics = TRUE'.</span></span>
<span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/build_SpatialExperiment.html">build_SpatialExperiment</a></span><span class="op">(</span></span>
<span>    <span class="va">sample_info</span>,</span>
<span>    coords_dir <span class="op">=</span> <span class="va">spe_input_dir</span>, reference_gtf <span class="op">=</span> <span class="va">gtf_cache</span>,</span>
<span>    calc_error_metrics <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Building SpatialExperiment using capture area as sample ID</span></span>
<span><span class="co">#&gt; 2024-10-16 19:27:45.859041 SpatialExperiment::read10xVisium: reading basic data from SpaceRanger</span></span>
<span><span class="co">#&gt; 2024-10-16 19:27:52.626383 read10xVisiumAnalysis: reading analysis output from SpaceRanger</span></span>
<span><span class="co">#&gt; 2024-10-16 19:27:53.017163 add10xVisiumAnalysis: adding analysis output from SpaceRanger</span></span>
<span><span class="co">#&gt; 2024-10-16 19:27:53.364326 rtracklayer::import: reading the reference GTF file</span></span>
<span><span class="co">#&gt; 2024-10-16 19:28:23.677907 adding gene information to the SPE object</span></span>
<span><span class="co">#&gt; Warning: Gene IDs did not match. This typically happens when you are not using</span></span>
<span><span class="co">#&gt; the same GTF file as the one that was used by SpaceRanger. For example, one</span></span>
<span><span class="co">#&gt; file uses GENCODE IDs and the other one ENSEMBL IDs. read10xVisiumWrapper()</span></span>
<span><span class="co">#&gt; will try to convert them to ENSEMBL IDs.</span></span>
<span><span class="co">#&gt; Warning: Dropping 2226 out of 38606 genes for which we don't have information</span></span>
<span><span class="co">#&gt; on the reference GTF file. This typically happens when you are not using the</span></span>
<span><span class="co">#&gt; same GTF file as the one that was used by SpaceRanger.</span></span>
<span><span class="co">#&gt; 2024-10-16 19:28:23.903427 adding information used by spatialLIBD</span></span>
<span><span class="co">#&gt; Overwriting imgData(spe) with merged images (one per group)</span></span>
<span><span class="co">#&gt; Adding array coordinates with error metrics and adding overlap info</span></span>
<span></span>
<span><span class="co">## The images in this example data have to be mirrored across the horizontal axis.</span></span>
<span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu">SpatialExperiment</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SpatialExperiment/man/SpatialExperiment-rotate-mirror.html" class="external-link">mirrorObject</a></span><span class="op">(</span><span class="va">spe</span>, axis <span class="op">=</span> <span class="st">"h"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Explore stitched spe object</span></span>
<span><span class="va">spe</span></span>
<span><span class="co">#&gt; class: SpatialExperiment </span></span>
<span><span class="co">#&gt; dim: 36380 14976 </span></span>
<span><span class="co">#&gt; metadata(0):</span></span>
<span><span class="co">#&gt; assays(1): counts</span></span>
<span><span class="co">#&gt; rownames(36380): ENSG00000243485.5 ENSG00000237613.2 ...</span></span>
<span><span class="co">#&gt;   ENSG00000198695.2 ENSG00000198727.2</span></span>
<span><span class="co">#&gt; rowData names(6): source type ... gene_type gene_search</span></span>
<span><span class="co">#&gt; colnames(14976): AAACAACGAATAGTTC-1_V13B23-283_A1</span></span>
<span><span class="co">#&gt;   AAACAAGTATCTCCCA-1_V13B23-283_A1 ... TTGTTTGTATTACACG-1_V13B23-283_D1</span></span>
<span><span class="co">#&gt;   TTGTTTGTGTAAATTC-1_V13B23-283_D1</span></span>
<span><span class="co">#&gt; colData names(33): sample_id in_tissue ... overlap_key</span></span>
<span><span class="co">#&gt;   exclude_overlapping</span></span>
<span><span class="co">#&gt; reducedDimNames(0):</span></span>
<span><span class="co">#&gt; mainExpName: NULL</span></span>
<span><span class="co">#&gt; altExpNames(0):</span></span>
<span><span class="co">#&gt; spatialCoords names(2) : pxl_col_in_fullres pxl_row_in_fullres</span></span>
<span><span class="co">#&gt; imgData names(4): sample_id image_id data scaleFactor</span></span></code></pre></div>
<p>The <code>colData(spe)$exclude_overlapping</code> column controls
which spots to drop for visualization purposes. Note also that the
<code>overlap_key</code> column was added, which gives a comma-separated
string of spot keys overlapping each given spot, or the empty string
otherwise. After spatial clustering, the <code>overlap_key</code>
information can be useful to check how frequently overlapping spots are
assigned the same cluster.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Examine spots to exclude for plotting</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">spe</span><span class="op">$</span><span class="va">exclude_overlapping</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; FALSE  TRUE </span></span>
<span><span class="co">#&gt; 13426  1550</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="examining-the-stitched-data">Examining the stitched data<a class="anchor" aria-label="anchor" href="#examining-the-stitched-data"></a>
</h2>
<div class="section level3">
<h3 id="stitched-plotting">Stitched plotting<a class="anchor" aria-label="anchor" href="#stitched-plotting"></a>
</h3>
<p>To demonstrate that we’ve stitched both the gene expression and image
data successfully, we’ll use
<code>spatialLIBD::vis_gene(is_stitched = TRUE)</code> (version 1.17.8
or newer) to plot the distribution of white matter spatially. For more
context on human brain white matter spatial marker genes, check <a href="https://doi.org/10.1038/s41593-020-00787-0" class="external-link">our previous work on
this subject</a>.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Show combined raw expression of white-matter marker genes</span></span>
<span><span class="va">wm_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">[</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/match.html" class="external-link">match</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"MBP"</span>, <span class="st">"GFAP"</span>, <span class="st">"PLP1"</span>, <span class="st">"AQP4"</span><span class="op">)</span>, <span class="fu">rowData</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">$</span><span class="va">gene_name</span><span class="op">)</span></span>
<span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/spatialLIBD/man/vis_gene.html" class="external-link">vis_gene</a></span><span class="op">(</span><span class="va">spe</span>, geneid <span class="op">=</span> <span class="va">wm_genes</span>, assayname <span class="op">=</span> <span class="st">"counts"</span>, is_stitched <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="visiumStitched_files/figure-html/explore_coords-1.png" width="700"></p>
<p>Note that we’re plotting raw counts; prior to normalization,
library-size variation across spots can bias the apparent distribution.
Later, we’ll show that normalization is critical to producing a visually
seamless transition between overlapping capture areas.</p>
</div>
<div class="section level3">
<h3 id="defining-array-coordinates">Defining Array Coordinates<a class="anchor" aria-label="anchor" href="#defining-array-coordinates"></a>
</h3>
<p>Given that the stitched data is larger than a default Visium capture
area, <code><a href="../reference/add_array_coords.html">add_array_coords()</a></code> (which is used internally by
<code><a href="../reference/build_SpatialExperiment.html">build_SpatialExperiment()</a></code>) recomputed the array coordinates
(i.e. <code>spe$array_row</code> and <code>spe$array_col</code>) to more
sensibly index the stitched data.</p>
<p>Let’s explain this in more detail. By definition, these array
coordinates (see <a href="https://www.10xgenomics.com/support/software/space-ranger/latest/analysis/outputs/spatial-outputs#tissue-positions" class="external-link">documentation
from 10X</a>) are integer indices of each spot on a Visium capture area,
numbering the typically 78 and 128 rows and columns, respectively, for a
6.5mm capture area. The <code><a href="../reference/build_SpatialExperiment.html">build_SpatialExperiment()</a></code> function
retains each capture area’s original array coordinates,
<code>spe$array_row_original</code> and
<code>spe$array_col_original</code>, but these are typically not useful
to represent our group-level, stitched data. In fact, each stitched
capture area has the same exact array coordinates, despite having
different spatial positions after stitching. We’ll take in-tissue spots
only and use transparency to emphasize the overlap among capture
areas:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Plot positions of default array coordinates, before overwriting with more</span></span>
<span><span class="co">## meaningful values. Use custom colors for each capture area</span></span>
<span><span class="va">ca_colors</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"#A33B20"</span>, <span class="st">"#e7bb41"</span>, <span class="st">"#3d3b8e"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">ca_colors</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"V13B23-283_C1"</span>, <span class="st">"V13B23-283_D1"</span>, <span class="st">"V13B23-283_A1"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">colData</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">in_tissue</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span></span>
<span>        mapping <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span></span>
<span>            x <span class="op">=</span> <span class="va">array_row_original</span>, y <span class="op">=</span> <span class="va">array_col_original</span>, color <span class="op">=</span> <span class="va">capture_area</span></span>
<span>        <span class="op">)</span></span>
<span>    <span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>alpha <span class="op">=</span> <span class="fl">0.3</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_color_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="va">ca_colors</span><span class="op">)</span></span></code></pre></div>
<p><img src="visiumStitched_files/figure-html/array_plot_orig-1.png" width="700"></p>
<p>Let’s contrast this with the array coordinates recomputed by
<code>visiumStitched</code>. Briefly, <code>visiumStitched</code> forms
a new hexagonal, Visium-like grid spanning the space occupied by all
capture areas after stitching. Then, the true spot positions are fit to
the nearest new spot positions, in terms of Euclidean distance. Finally,
array coordinates are re-indexed according to the new spot assignments,
resulting in spatially meaningful values that apply at the group level
for stitched data.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Plot positions of redefined array coordinates</span></span>
<span><span class="fu">colData</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">in_tissue</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span></span>
<span>        mapping <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span></span>
<span>            x <span class="op">=</span> <span class="va">array_row</span>, y <span class="op">=</span> <span class="va">array_col</span>, color <span class="op">=</span> <span class="va">capture_area</span></span>
<span>        <span class="op">)</span></span>
<span>    <span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>alpha <span class="op">=</span> <span class="fl">0.3</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_color_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="va">ca_colors</span><span class="op">)</span></span></code></pre></div>
<p><img src="visiumStitched_files/figure-html/array_plot_after-1.png" width="700"></p>
<p>An important downstream application of these array coordinates, is
that it enables methods that rely on the hexagonal grid structure of
Visium to find more than the original six neighboring spots. This
enables clustering with <a href="https://doi.org/10.1038/s41587-021-00935-2" class="external-link"><code>BayesSpace</code></a>
or <a href="https://doi.org/10.1038/s41467-023-35947-w" class="external-link"><code>PRECAST</code></a>,
to treat each group as a spatially continuous sample. We can see here
how <a href="https://github.com/edward130603/BayesSpace/blob/8e9af8f2fa8e93518cf9ecee1ded9ab93e88fffd/R/spatialCluster.R#L214-L220" class="external-link"><code>BayesSpace:::.find_neighbors()</code></a>
version 1.11.0 uses the hexagonal Visium grid properties to find the
spot neighbors. See also <a href="https://www.nature.com/articles/s41587-021-00935-2/figures/1" class="external-link"><code>BayesSpace</code>
Figure 1b</a> for an illustration of this process.</p>
<iframe frameborder="0" scrolling="no" style="width:100%; height:226px;" allow="clipboard-write" src="https://emgithub.com/iframe.html?target=https%3A%2F%2Fgithub.com%2Fedward130603%2FBayesSpace%2Fblob%2F8e9af8f2fa8e93518cf9ecee1ded9ab93e88fffd%2FR%2FspatialCluster.R%23L214-L220&amp;style=default&amp;type=code&amp;showBorder=on&amp;showLineNumbers=on&amp;showFileMeta=on&amp;showFullPath=on&amp;showCopy=on">
</iframe>
<p>Yet, it doesn’t matter if there are actually two or more spots on
each of those six neighbor positions. <code>visiumStitched</code> takes
advantage of this property to enable <code>BayesSpace</code> and other
spatially-aware clustering methods to use data from overlapping spots
when performing spatial clustering. You can then use
<code>colData(spe)$overlap_key</code> to inspect whether overlapping
spots were assigned to the same spatial cluster.</p>
<div class="section level4">
<h4 id="error-metrics">Error metrics<a class="anchor" aria-label="anchor" href="#error-metrics"></a>
</h4>
<p>No algorithm can fit a set of capture areas’ spots onto a single
hexagonal grid without some error. Here, we define a spot’s error in
being assigned new array coordinates with two independent metrics, which
are stored in <code>spe$euclidean_error</code> and
<code>spe$shared_neighbors</code> if the user opts to compute them with
<code>build_SpatialExperiment(calc_error_metrics = TRUE)</code>. The
latter metric can take a couple minutes to compute, and thus by default
the metrics are not computed.</p>
<p>The first metric is the Euclidean distance, in multiples of 100
microns (the distance between spots on a Visium capture area), between a
spot’s original position and the position of its assigned array
coordinates.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#   Explore the distribution of Euclidean error</span></span>
<span><span class="fu">colData</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span>mapping <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="fl">0</span>, y <span class="op">=</span> <span class="va">euclidean_error</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html" class="external-link">geom_boxplot</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="visiumStitched_files/figure-html/euclidean_error-1.png" width="700"></p>
<p>The other metric, <code>spe$shared_neighbors</code>, measures the
fraction of original neighbors (from a same capture area) that are
retained after mapping to the new array coordinates. Thus, a value of 1
is ideal.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#   Explore the distribution of Euclidean error</span></span>
<span><span class="fu">colData</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span>mapping <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="fl">0</span>, y <span class="op">=</span> <span class="va">shared_neighbors</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html" class="external-link">geom_boxplot</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="visiumStitched_files/figure-html/shared_neighbors-1.png" width="700"></p>
<p>In theory, error as measured through these metrics could have a very
slight impact on the quality of clustering results downstream. We
envision interested users in checking these metrics when interpreting
specific spots’ cluster assignments downstream.</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="downstream-applications">Downstream applications<a class="anchor" aria-label="anchor" href="#downstream-applications"></a>
</h2>
<p>One common area of analysis in spatial transcriptomics involves
clustering– in particular, spatially-aware clustering. Many
spatially-aware clustering algorithms check the array coordinates to
determine neighboring spots and ultimately produce spatially smooth
clusters. As we have previously explained, <code>visiumStitched</code>
<a href="#array-coordinates">re-computes array coordinates</a> in a
meaningful way, such that software like <a href="https://doi.org/10.1038/s41587-021-00935-2" class="external-link"><code>BayesSpace</code></a>
and <a href="https://doi.org/10.1038/s41467-023-35947-w" class="external-link"><code>PRECAST</code></a>
work out-of-the-box with stitched data, treating each group as a single
continuous sample.</p>
<p><a href="https://github.com/LieberInstitute/visiumStitched_brain/blob/devel/code/03_stitching/06_precast.R" class="external-link">We’ve
already run PRECAST</a>, and can visualize the results here, where we
see a fairly seamless transition of cluster assignments across
capture-area boundaries. First, let’s examine <code>k = 2</code>:</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Grab SpatialExperiment with normalized counts</span></span>
<span><span class="va">spe_norm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/spatialLIBD/man/fetch_data.html" class="external-link">fetch_data</a></span><span class="op">(</span>type <span class="op">=</span> <span class="st">"visiumStitched_brain_spe"</span><span class="op">)</span></span>
<span><span class="co">#&gt; 2024-10-16 19:29:38.509273 loading file /github/home/.cache/R/BiocFileCache/1c756c878402_visiumStitched_brain_spe.rds%3Frlkey%3Dnq6a82u23xuu9hohr86oodwdi%26dl%3D1</span></span>
<span><span class="fu">assayNames</span><span class="op">(</span><span class="va">spe_norm</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "counts"    "logcounts"</span></span>
<span></span>
<span><span class="co">## PRECAST k = 2 clusters with our manually chosen colors</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/spatialLIBD/man/vis_clus.html" class="external-link">vis_clus</a></span><span class="op">(</span></span>
<span>    <span class="va">spe_norm</span>,</span>
<span>    clustervar <span class="op">=</span> <span class="st">"precast_k2_stitched"</span>,</span>
<span>    is_stitched <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    colors <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>        <span class="st">"1"</span> <span class="op">=</span> <span class="st">"gold"</span>,</span>
<span>        <span class="st">"2"</span> <span class="op">=</span> <span class="st">"darkblue"</span>,</span>
<span>        <span class="st">"NA"</span> <span class="op">=</span> <span class="st">"white"</span></span>
<span>    <span class="op">)</span>,</span>
<span>    spatial <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="visiumStitched_files/figure-html/precast_k2_stitched-1.png" width="700"></p>
<p>We can see that these two spatial clusters are differentiating the
white vs the gray matter based on the white matter marker genes we <a href="https://research.libd.org/visiumStitched/articles/visiumStitched.html#a-note-on-normalization" class="external-link">previously
visualized</a>.</p>
<p>In the example data, <code>k = 4</code> and <code>k = 8</code> have
also been computed. Let’s visualize the <code>k = 4</code> results.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## PRECAST results already available in this example data</span></span>
<span><span class="va">vars</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">spe_norm</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">vars</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"precast"</span>, <span class="va">vars</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="co">#&gt;  [1] "precast_k2_stitched"    "precast_k4_stitched"    "precast_k8_stitched"   </span></span>
<span><span class="co">#&gt;  [4] "precast_k16_stitched"   "precast_k24_stitched"   "precast_k2_unstitched" </span></span>
<span><span class="co">#&gt;  [7] "precast_k4_unstitched"  "precast_k8_unstitched"  "precast_k16_unstitched"</span></span>
<span><span class="co">#&gt; [10] "precast_k24_unstitched"</span></span>
<span></span>
<span><span class="co">## PRECAST k = 4 clusters with default cluster colors</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/spatialLIBD/man/vis_clus.html" class="external-link">vis_clus</a></span><span class="op">(</span></span>
<span>    <span class="va">spe_norm</span>,</span>
<span>    clustervar <span class="op">=</span> <span class="st">"precast_k4_stitched"</span>,</span>
<span>    is_stitched <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    spatial <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="visiumStitched_files/figure-html/precast_k4_stitched-1.png" width="700"></p>
<p>The biological interpretation of these spatial clusters would need
further work, using methods such as:</p>
<ul>
<li>
<a href="https://research.libd.org/spatialLIBD/articles/guide_to_spatial_registration.html" class="external-link">spatial
registration</a> of reference sc/snRNA-seq or spatial data,</li>
<li>visualization of known marker genes for the tissue of interest,</li>
<li>or identification of data driven marker genes using <a href="https://research.libd.org/spatialLIBD/reference/registration_wrapper.html" class="external-link"><code>spatialLIBD::registration_wrapper()</code></a>,
<a href="https://research.libd.org/DeconvoBuddies/reference/findMarkers_1vAll.html" class="external-link"><code>DeconvoBuddies::findMarkers_1vAll()</code></a>,
<a href="https://research.libd.org/DeconvoBuddies/reference/get_mean_ratio.html" class="external-link"><code>DeconvoBuiddies::get_mean_ratio()</code></a>
or other tools. See <a href="https://doi.org/10.1186/s13059-024-03183-0" class="external-link">Pullin and McCarthy,
<em>Genome Biol.</em>, 2024</a> for a list of marker gene selection
methods.</li>
</ul>
</div>
<div class="section level2">
<h2 id="conclusion">Conclusion<a class="anchor" aria-label="anchor" href="#conclusion"></a>
</h2>
<p><code>visiumStitched</code> provides a set of helper functions, in
conjunction with <code>ImageJ</code>/<code>Fiji</code>, intended to
simplify the stitching of Visium data into a spatially integrated
<code>SpatialExperiment</code> object ready for analysis. We hope you
find it useful for your research!</p>
</div>
<div class="section level2">
<h2 id="reproducibility">Reproducibility<a class="anchor" aria-label="anchor" href="#reproducibility"></a>
</h2>
<p>The <em><a href="https://bioconductor.org/packages/3.20/visiumStitched" class="external-link">visiumStitched</a></em>
package (Eagles and Collado-Torres, 2024) was made possible thanks
to:</p>
<ul>
<li>R (R Core Team, 2024)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.20/BiocFileCache" class="external-link">BiocFileCache</a></em>
(Shepherd and Morgan, 2024)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.20/BiocStyle" class="external-link">BiocStyle</a></em>
(Oleś, 2024)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=dplyr" class="external-link">dplyr</a></em> (Wickham,
François, Henry, Müller, and Vaughan, 2023)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.20/DropletUtils" class="external-link">DropletUtils</a></em>
(Lun, Riesenfeld, Andrews, Dao, Gomes, participants in the 1st Human
Cell Atlas Jamboree, and Marioni, 2019)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=ggplot2" class="external-link">ggplot2</a></em>
(Wickham, 2016)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=imager" class="external-link">imager</a></em>
(Barthelme, 2024)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=knitr" class="external-link">knitr</a></em> (Xie,
2024)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=pkgcond" class="external-link">pkgcond</a></em> (Redd
and R Documentation Task Force, 2021)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=RefManageR" class="external-link">RefManageR</a></em>
(McLean, 2017)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=rjson" class="external-link">rjson</a></em>
(Couture-Beil, 2024)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=rmarkdown" class="external-link">rmarkdown</a></em>
(Allaire, Xie, Dervieux, McPherson, Luraschi, Ushey, Atkins, Wickham,
Cheng, Chang, and Iannone, 2024)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.20/S4Vectors" class="external-link">S4Vectors</a></em>
(Pagès, Lawrence, and Aboyoun, 2024)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=sessioninfo" class="external-link">sessioninfo</a></em>
(Wickham, Chang, Flight, Müller, and Hester, 2021)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=Seurat" class="external-link">Seurat</a></em> (Hao,
Stuart, Kowalski, Choudhary, Hoffman, Hartman, Srivastava, Molla, Madad,
Fernandez-Granda, and Satija, 2023)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.20/SpatialExperiment" class="external-link">SpatialExperiment</a></em>
(Righelli, Weber, Crowell, Pardo, Collado-Torres, Ghazanfar, Lun, Hicks,
and Risso, 2022)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.20/spatialLIBD" class="external-link">spatialLIBD</a></em>
(Pardo, Spangler, Weber, Hicks, Jaffe, Martinowich, Maynard, and
Collado-Torres, 2022)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=stringr" class="external-link">stringr</a></em>
(Wickham, 2023)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.20/SummarizedExperiment" class="external-link">SummarizedExperiment</a></em>
(Morgan, Obenchain, Hester, and Pagès, 2024)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=testthat" class="external-link">testthat</a></em>
(Wickham, 2011)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=xml2" class="external-link">xml2</a></em>
(Wickham, Hester, and Ooms, 2023)</li>
</ul>
<p>This package was developed using <em><a href="https://bioconductor.org/packages/3.20/biocthis" class="external-link">biocthis</a></em>.</p>
<p>Code for creating the vignette</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Create the vignette</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/rstudio/rmarkdown" class="external-link">"rmarkdown"</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/system.time.html" class="external-link">system.time</a></span><span class="op">(</span><span class="fu"><a href="https://pkgs.rstudio.com/rmarkdown/reference/render.html" class="external-link">render</a></span><span class="op">(</span><span class="st">"visiumStitched.Rmd"</span>, <span class="st">"BiocStyle::html_document"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Extract the R code</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://yihui.org/knitr/" class="external-link">"knitr"</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/knit.html" class="external-link">knit</a></span><span class="op">(</span><span class="st">"visiumStitched.Rmd"</span>, tangle <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><code>R</code> session information.</p>
<pre><code><span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt;  <span style="color: #555555; font-style: italic;">setting </span> <span style="color: #555555; font-style: italic;">value</span></span></span>
<span><span class="co">#&gt;  version  R version 4.4.1 (2024-06-14)</span></span>
<span><span class="co">#&gt;  os       Ubuntu 22.04.5 LTS</span></span>
<span><span class="co">#&gt;  system   x86_64, linux-gnu</span></span>
<span><span class="co">#&gt;  ui       X11</span></span>
<span><span class="co">#&gt;  language en</span></span>
<span><span class="co">#&gt;  collate  en_US.UTF-8</span></span>
<span><span class="co">#&gt;  ctype    en_US.UTF-8</span></span>
<span><span class="co">#&gt;  tz       UTC</span></span>
<span><span class="co">#&gt;  date     2024-10-16</span></span>
<span><span class="co">#&gt;  pandoc   3.4 @ /usr/bin/ (via rmarkdown)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt;  <span style="color: #555555; font-style: italic;">package             </span> <span style="color: #555555; font-style: italic;">*</span> <span style="color: #555555; font-style: italic;">version   </span> <span style="color: #555555; font-style: italic;">date (UTC)</span> <span style="color: #555555; font-style: italic;">lib</span> <span style="color: #555555; font-style: italic;">source</span></span></span>
<span><span class="co">#&gt;  abind                  1.4-8      <span style="color: #555555;">2024-09-12</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  AnnotationDbi          1.67.0     <span style="color: #555555;">2024-05-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  AnnotationHub          3.13.3     <span style="color: #555555;">2024-08-19</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  attempt                0.3.1      <span style="color: #555555;">2020-05-03</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  backports              1.5.0      <span style="color: #555555;">2024-05-23</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  beachmat               2.21.6     <span style="color: #555555;">2024-09-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  beeswarm               0.4.0      <span style="color: #555555;">2021-06-01</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  benchmarkme            1.0.8      <span style="color: #555555;">2022-06-12</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  benchmarkmeData        1.0.4      <span style="color: #555555;">2020-04-23</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  bibtex                 0.5.1      <span style="color: #555555;">2023-01-26</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  Biobase              * 2.65.1     <span style="color: #555555;">2024-08-28</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  BiocBaseUtils          1.7.3      <span style="color: #555555;">2024-08-29</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  BiocFileCache        * 2.13.2     <span style="color: #555555;">2024-10-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  BiocGenerics         * 0.51.3     <span style="color: #555555;">2024-10-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  BiocIO                 1.15.2     <span style="color: #555555;">2024-08-23</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  BiocManager            1.30.25    <span style="color: #555555;">2024-08-28</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  BiocNeighbors          1.99.2     <span style="color: #555555;">2024-10-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  BiocParallel           1.39.0     <span style="color: #555555;">2024-05-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  BiocSingular           1.21.4     <span style="color: #555555;">2024-09-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  BiocStyle            * 2.33.1     <span style="color: #555555;">2024-06-12</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  BiocVersion            3.20.0     <span style="color: #555555;">2024-05-01</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  Biostrings             2.73.2     <span style="color: #555555;">2024-09-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  bit                    4.5.0      <span style="color: #555555;">2024-09-20</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  bit64                  4.5.2      <span style="color: #555555;">2024-09-22</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  bitops                 1.0-9      <span style="color: #555555;">2024-10-03</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  blob                   1.2.4      <span style="color: #555555;">2023-03-17</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  bmp                    0.3        <span style="color: #555555;">2017-09-11</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  bookdown               0.41       <span style="color: #555555;">2024-10-16</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  bslib                  0.8.0      <span style="color: #555555;">2024-07-29</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  cachem                 1.1.0      <span style="color: #555555;">2024-05-16</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  cli                    3.6.3      <span style="color: #555555;">2024-06-21</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  codetools              0.2-20     <span style="color: #555555;">2024-03-31</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  colorspace             2.1-1      <span style="color: #555555;">2024-07-26</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  config                 0.3.2      <span style="color: #555555;">2023-08-30</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  cowplot                1.1.3      <span style="color: #555555;">2024-01-22</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  crayon                 1.5.3      <span style="color: #555555;">2024-06-20</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  curl                   5.2.3      <span style="color: #555555;">2024-09-20</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  data.table             1.16.2     <span style="color: #555555;">2024-10-10</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  DBI                    1.2.3      <span style="color: #555555;">2024-06-02</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  dbplyr               * 2.5.0      <span style="color: #555555;">2024-03-19</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  DelayedArray           0.31.14    <span style="color: #555555;">2024-10-03</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  DelayedMatrixStats     1.27.3     <span style="color: #555555;">2024-08-08</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  desc                   1.4.3      <span style="color: #555555;">2023-12-10</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  digest                 0.6.37     <span style="color: #555555;">2024-08-19</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  doParallel             1.0.17     <span style="color: #555555;">2022-02-07</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  dotCall64              1.2        <span style="color: #555555;">2024-10-04</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  dplyr                * 1.1.4      <span style="color: #555555;">2023-11-17</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  dqrng                  0.4.1      <span style="color: #555555;">2024-05-28</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  DropletUtils           1.25.2     <span style="color: #555555;">2024-07-23</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  DT                     0.33       <span style="color: #555555;">2024-04-04</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  edgeR                  4.3.19     <span style="color: #555555;">2024-10-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  evaluate               1.0.1      <span style="color: #555555;">2024-10-10</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  ExperimentHub          2.13.1     <span style="color: #555555;">2024-07-31</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  fansi                  1.0.6      <span style="color: #555555;">2023-12-08</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  farver                 2.1.2      <span style="color: #555555;">2024-05-13</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  fastmap                1.2.0      <span style="color: #555555;">2024-05-15</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  fields                 16.3       <span style="color: #555555;">2024-09-30</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  filelock               1.0.3      <span style="color: #555555;">2023-12-11</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  foreach                1.5.2      <span style="color: #555555;">2022-02-02</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  fs                     1.6.4      <span style="color: #555555;">2024-04-25</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  generics               0.1.3      <span style="color: #555555;">2022-07-05</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  GenomeInfoDb         * 1.41.2     <span style="color: #555555;">2024-10-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  GenomeInfoDbData       1.2.13     <span style="color: #555555;">2024-10-16</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  GenomicAlignments      1.41.0     <span style="color: #555555;">2024-05-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  GenomicRanges        * 1.57.2     <span style="color: #555555;">2024-10-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  ggbeeswarm             0.7.2      <span style="color: #555555;">2023-04-29</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  ggplot2              * 3.5.1      <span style="color: #555555;">2024-04-23</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  ggrepel                0.9.6      <span style="color: #555555;">2024-09-07</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  glue                   1.8.0      <span style="color: #555555;">2024-09-30</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  golem                  0.5.1      <span style="color: #555555;">2024-08-27</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  gridExtra              2.3        <span style="color: #555555;">2017-09-09</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  gtable                 0.3.5      <span style="color: #555555;">2024-04-22</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  HDF5Array              1.33.8     <span style="color: #555555;">2024-10-04</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  highr                  0.11       <span style="color: #555555;">2024-05-26</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  hms                    1.1.3      <span style="color: #555555;">2023-03-21</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  htmltools              0.5.8.1    <span style="color: #555555;">2024-04-04</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  htmlwidgets            1.6.4      <span style="color: #555555;">2023-12-06</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  httpuv                 1.6.15     <span style="color: #555555;">2024-03-26</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  httr                   1.4.7      <span style="color: #555555;">2023-08-15</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  igraph                 2.0.3      <span style="color: #555555;">2024-03-13</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  imager                 1.0.2      <span style="color: #555555;">2024-05-13</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  IRanges              * 2.39.2     <span style="color: #555555;">2024-07-17</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  irlba                  2.3.5.1    <span style="color: #555555;">2022-10-03</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  iterators              1.0.14     <span style="color: #555555;">2022-02-05</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  jpeg                   0.1-10     <span style="color: #555555;">2022-11-29</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  jquerylib              0.1.4      <span style="color: #555555;">2021-04-26</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  jsonlite               1.8.9      <span style="color: #555555;">2024-09-20</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  KEGGREST               1.45.1     <span style="color: #555555;">2024-06-17</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  knitr                  1.48       <span style="color: #555555;">2024-07-07</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  labeling               0.4.3      <span style="color: #555555;">2023-08-29</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  later                  1.3.2      <span style="color: #555555;">2023-12-06</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  lattice                0.22-6     <span style="color: #555555;">2024-03-20</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  lazyeval               0.2.2      <span style="color: #555555;">2019-03-15</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  lifecycle              1.0.4      <span style="color: #555555;">2023-11-07</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  limma                  3.61.12    <span style="color: #555555;">2024-09-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  locfit                 1.5-9.10   <span style="color: #555555;">2024-06-24</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  lubridate              1.9.3      <span style="color: #555555;">2023-09-27</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  magick                 2.8.5      <span style="color: #555555;">2024-09-20</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  magrittr               2.0.3      <span style="color: #555555;">2022-03-30</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  maps                   3.4.2      <span style="color: #555555;">2023-12-15</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  Matrix                 1.7-0      <span style="color: #555555;">2024-04-26</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  MatrixGenerics       * 1.17.0     <span style="color: #555555;">2024-05-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  matrixStats          * 1.4.1      <span style="color: #555555;">2024-09-08</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  memoise                2.0.1      <span style="color: #555555;">2021-11-26</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  mime                   0.12       <span style="color: #555555;">2021-09-28</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  munsell                0.5.1      <span style="color: #555555;">2024-04-01</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  paletteer              1.6.0      <span style="color: #555555;">2024-01-21</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  pillar                 1.9.0      <span style="color: #555555;">2023-03-22</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  pkgcond                0.1.1      <span style="color: #555555;">2021-04-28</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  pkgconfig              2.0.3      <span style="color: #555555;">2019-09-22</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  pkgdown                <span style="color: #BB00BB; font-weight: bold;">2.1.1.9000</span> <span style="color: #555555;">2024-10-16</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">Github (r-lib/pkgdown@71fd32c)</span></span></span>
<span><span class="co">#&gt;  plotly                 4.10.4     <span style="color: #555555;">2024-01-13</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  plyr                   1.8.9      <span style="color: #555555;">2023-10-02</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  png                    0.1-8      <span style="color: #555555;">2022-11-29</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  promises               1.3.0      <span style="color: #555555;">2024-04-05</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  purrr                  1.0.2      <span style="color: #555555;">2023-08-10</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  R.methodsS3            1.8.2      <span style="color: #555555;">2022-06-13</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  R.oo                   1.26.0     <span style="color: #555555;">2024-01-24</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  R.utils                2.12.3     <span style="color: #555555;">2023-11-18</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  R6                     2.5.1      <span style="color: #555555;">2021-08-19</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  ragg                   1.3.3      <span style="color: #555555;">2024-09-11</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  rappdirs               0.3.3      <span style="color: #555555;">2021-01-31</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  RColorBrewer           1.1-3      <span style="color: #555555;">2022-04-03</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  Rcpp                   1.0.13     <span style="color: #555555;">2024-07-17</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  RCurl                  1.98-1.16  <span style="color: #555555;">2024-07-11</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  readbitmap             0.1.5      <span style="color: #555555;">2018-06-27</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  readr                  2.1.5      <span style="color: #555555;">2024-01-10</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  RefManageR           * 1.4.0      <span style="color: #555555;">2022-09-30</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  rematch2               2.1.2      <span style="color: #555555;">2020-05-01</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  restfulr               0.0.15     <span style="color: #555555;">2022-06-16</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  rhdf5                  2.49.0     <span style="color: #555555;">2024-05-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  rhdf5filters           1.17.0     <span style="color: #555555;">2024-05-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  Rhdf5lib               1.27.0     <span style="color: #555555;">2024-05-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  rjson                  0.2.23     <span style="color: #555555;">2024-09-16</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  rlang                  1.1.4      <span style="color: #555555;">2024-06-04</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  rmarkdown              2.28       <span style="color: #555555;">2024-08-17</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  Rsamtools              2.21.2     <span style="color: #555555;">2024-09-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  RSQLite                2.3.7      <span style="color: #555555;">2024-05-27</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  rsvd                   1.0.5      <span style="color: #555555;">2021-04-16</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  rtracklayer            1.65.0     <span style="color: #555555;">2024-05-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  S4Arrays               1.5.11     <span style="color: #555555;">2024-10-14</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  S4Vectors            * 0.43.2     <span style="color: #555555;">2024-07-17</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  sass                   0.4.9      <span style="color: #555555;">2024-03-15</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  ScaledMatrix           1.13.0     <span style="color: #555555;">2024-05-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  scales                 1.3.0      <span style="color: #555555;">2023-11-28</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  scater                 1.33.4     <span style="color: #555555;">2024-07-21</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  scuttle                1.15.4     <span style="color: #555555;">2024-08-14</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  sessioninfo          * 1.2.2      <span style="color: #555555;">2021-12-06</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  shiny                  1.9.1      <span style="color: #555555;">2024-08-01</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  shinyWidgets           0.8.7      <span style="color: #555555;">2024-09-23</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  SingleCellExperiment * 1.27.2     <span style="color: #555555;">2024-05-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  spam                   2.11-0     <span style="color: #555555;">2024-10-03</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  SparseArray            1.5.44     <span style="color: #555555;">2024-10-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  sparseMatrixStats      1.17.2     <span style="color: #555555;">2024-06-12</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  SpatialExperiment    * 1.15.1     <span style="color: #555555;">2024-06-20</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  spatialLIBD          * 1.17.8     <span style="color: #555555;">2024-10-16</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">Github (LieberInstitute/spatialLIBD@c82c789)</span></span></span>
<span><span class="co">#&gt;  statmod                1.5.0      <span style="color: #555555;">2023-01-06</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  stringi                1.8.4      <span style="color: #555555;">2024-05-06</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  stringr                1.5.1      <span style="color: #555555;">2023-11-14</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  SummarizedExperiment * 1.35.4     <span style="color: #555555;">2024-10-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  systemfonts            1.1.0      <span style="color: #555555;">2024-05-15</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  textshaping            0.4.0      <span style="color: #555555;">2024-05-24</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  tibble                 3.2.1      <span style="color: #555555;">2023-03-20</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  tidyr                  1.3.1      <span style="color: #555555;">2024-01-24</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  tidyselect             1.2.1      <span style="color: #555555;">2024-03-11</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  tiff                   0.1-12     <span style="color: #555555;">2023-11-28</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  timechange             0.3.0      <span style="color: #555555;">2024-01-18</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  tzdb                   0.4.0      <span style="color: #555555;">2023-05-12</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  UCSC.utils             1.1.0      <span style="color: #555555;">2024-05-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  utf8                   1.2.4      <span style="color: #555555;">2023-10-22</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  vctrs                  0.6.5      <span style="color: #555555;">2023-12-01</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  vipor                  0.4.7      <span style="color: #555555;">2023-12-18</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  viridis                0.6.5      <span style="color: #555555;">2024-01-29</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  viridisLite            0.4.2      <span style="color: #555555;">2023-05-02</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  visiumStitched       * 0.99.8     <span style="color: #555555;">2024-10-16</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  vroom                  1.6.5      <span style="color: #555555;">2023-12-05</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  withr                  3.0.1      <span style="color: #555555;">2024-07-31</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  xfun                   0.48       <span style="color: #555555;">2024-10-03</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  XML                    3.99-0.17  <span style="color: #555555;">2024-06-25</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  xml2                   1.3.6      <span style="color: #555555;">2023-12-04</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  xtable                 1.8-4      <span style="color: #555555;">2019-04-21</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  XVector                0.45.0     <span style="color: #555555;">2024-05-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  yaml                   2.3.10     <span style="color: #555555;">2024-07-26</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  zlibbioc               1.51.1     <span style="color: #555555;">2024-06-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [1] /__w/_temp/Library</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [2] /usr/local/lib/R/site-library</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [3] /usr/local/lib/R/library</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></span></code></pre>
</div>
<div class="section level2">
<h2 id="bibliography">Bibliography<a class="anchor" aria-label="anchor" href="#bibliography"></a>
</h2>
<p>This vignette was generated using <em><a href="https://bioconductor.org/packages/3.20/BiocStyle" class="external-link">BiocStyle</a></em>
(Oleś, 2024) with <em><a href="https://CRAN.R-project.org/package=knitr" class="external-link">knitr</a></em> (Xie,
2024) and <em><a href="https://CRAN.R-project.org/package=rmarkdown" class="external-link">rmarkdown</a></em>
(Allaire, Xie, Dervieux et al., 2024) running behind the scenes.</p>
<p>Citations made with <em><a href="https://CRAN.R-project.org/package=RefManageR" class="external-link">RefManageR</a></em>
(McLean, 2017).</p>
<p>
<a id="bib-allaire2024rmarkdown"></a><a href="#cite-allaire2024rmarkdown">[1]</a><cite>
J. Allaire, Y. Xie, C. Dervieux, et al. <em>rmarkdown: Dynamic Documents
for R</em>. R package version 2.28. 2024. URL:
<a href="https://github.com/rstudio/rmarkdown" class="external-link">https://github.com/rstudio/rmarkdown</a>.</cite>
</p>
<p>
<a id="bib-barthelme2024imager"></a><a href="#cite-barthelme2024imager">[2]</a><cite>
S. Barthelme. <em>imager: Image Processing Library Based on ‘CImg’</em>.
R package version 1.0.2, <a href="https://github.com/asgr/imager/" class="external-link uri">https://github.com/asgr/imager/</a>. 2024. URL:
<a href="https://asgr.github.io/imager/" class="external-link">https://asgr.github.io/imager/</a>.</cite>
</p>
<p>
<a id="bib-couturebeil2024rjson"></a><a href="#cite-couturebeil2024rjson">[3]</a><cite>
A. Couture-Beil. <em>rjson: JSON for R</em>. R package version 0.2.23.
2024. URL:
<a href="https://github.com/alexcb/rjson" class="external-link">https://github.com/alexcb/rjson</a>.</cite>
</p>
<p>
<a id="bib-eagles2024enable"></a><a href="#cite-eagles2024enable">[4]</a><cite>
N. J. Eagles and L. Collado-Torres. <em>Enable downstream analysis of
Visium capture areas stitched together with Fiji</em>. <a href="https://github.com/LieberInstitute/visiumStitched/visiumStitched" class="external-link uri">https://github.com/LieberInstitute/visiumStitched/visiumStitched</a>
- R package version 0.99.8. 2024. DOI:
<a href="https://doi.org/10.18129/B9.bioc.visiumStitched" class="external-link">10.18129/B9.bioc.visiumStitched</a>.
URL:
<a href="http://www.bioconductor.org/packages/visiumStitched" class="external-link">http://www.bioconductor.org/packages/visiumStitched</a>.</cite>
</p>
<p>
<a id="bib-hao2023dictionary"></a><a href="#cite-hao2023dictionary">[5]</a><cite>
Y. Hao, T. Stuart, M. H. Kowalski, et al. “Dictionary learning for
integrative, multimodal and scalable single-cell analysis”. In:
<em>Nature Biotechnology</em> (2023). DOI:
<a href="https://doi.org/10.1038/s41587-023-01767-y" class="external-link">10.1038/s41587-023-01767-y</a>.
URL:
<a href="https://doi.org/10.1038/s41587-023-01767-y" class="external-link">https://doi.org/10.1038/s41587-023-01767-y</a>.</cite>
</p>
<p>
<a id="bib-lun2019emptydrops"></a><a href="#cite-lun2019emptydrops">[6]</a><cite>
A. T. L. Lun, S. Riesenfeld, T. Andrews, et al. “EmptyDrops:
distinguishing cells from empty droplets in droplet-based single-cell
RNA sequencing data”. In: <em>Genome Biol.</em> 20 (2019), p. 63. DOI:
<a href="https://doi.org/10.1186/s13059-019-1662-y" class="external-link">10.1186/s13059-019-1662-y</a>.</cite>
</p>
<p>
<a id="bib-mclean2017refmanager"></a><a href="#cite-mclean2017refmanager">[7]</a><cite>
M. W. McLean. “RefManageR: Import and Manage BibTeX and BibLaTeX
References in R”. In: <em>The Journal of Open Source Software</em>
(2017). DOI:
<a href="https://doi.org/10.21105/joss.00338" class="external-link">10.21105/joss.00338</a>.</cite>
</p>
<p>
<a id="bib-morgan2024summarizedexperiment"></a><a href="#cite-morgan2024summarizedexperiment">[8]</a><cite>
M. Morgan, V. Obenchain, J. Hester, et al. <em>SummarizedExperiment:
SummarizedExperiment container</em>. R package version 1.35.4. 2024.
DOI:
<a href="https://doi.org/10.18129/B9.bioc.SummarizedExperiment" class="external-link">10.18129/B9.bioc.SummarizedExperiment</a>.
URL:
<a href="https://bioconductor.org/packages/SummarizedExperiment" class="external-link">https://bioconductor.org/packages/SummarizedExperiment</a>.</cite>
</p>
<p>
<a id="bib-ole2024biocstyle"></a><a href="#cite-ole2024biocstyle">[9]</a><cite>
A. Oleś. <em>BiocStyle: Standard styles for vignettes and other
Bioconductor documents</em>. R package version 2.33.1. 2024. DOI:
<a href="https://doi.org/10.18129/B9.bioc.BiocStyle" class="external-link">10.18129/B9.bioc.BiocStyle</a>.
URL:
<a href="https://bioconductor.org/packages/BiocStyle" class="external-link">https://bioconductor.org/packages/BiocStyle</a>.</cite>
</p>
<p>
<a id="bib-pags2024s4vectors"></a><a href="#cite-pags2024s4vectors">[10]</a><cite>
H. Pagès, M. Lawrence, and P. Aboyoun. <em>S4Vectors: Foundation of
vector-like and list-like containers in Bioconductor</em>. R package
version 0.43.2. 2024. DOI:
<a href="https://doi.org/10.18129/B9.bioc.S4Vectors" class="external-link">10.18129/B9.bioc.S4Vectors</a>.
URL:
<a href="https://bioconductor.org/packages/S4Vectors" class="external-link">https://bioconductor.org/packages/S4Vectors</a>.</cite>
</p>
<p>
<a id="bib-pardo2022spatiallibd"></a><a href="#cite-pardo2022spatiallibd">[11]</a><cite>
B. Pardo, A. Spangler, L. M. Weber, et al. “spatialLIBD: an
R/Bioconductor package to visualize spatially-resolved transcriptomics
data”. In: <em>BMC Genomics</em> (2022). DOI:
<a href="https://doi.org/10.1186/s12864-022-08601-w" class="external-link">10.1186/s12864-022-08601-w</a>.
URL:
<a href="https://doi.org/10.1186/s12864-022-08601-w" class="external-link">https://doi.org/10.1186/s12864-022-08601-w</a>.</cite>
</p>
<p>
<a id="bib-2024language"></a><a href="#cite-2024language">[12]</a><cite>
R Core Team. <em>R: A Language and Environment for Statistical
Computing</em>. R Foundation for Statistical Computing. Vienna, Austria,
2024. URL:
<a href="https://www.R-project.org/" class="external-link">https://www.R-project.org/</a>.</cite>
</p>
<p>
<a id="bib-redd2021pkgcond"></a><a href="#cite-redd2021pkgcond">[13]</a><cite>
A. Redd and R Documentation Task Force. <em>pkgcond: Classed Error and
Warning Conditions</em>. R package version 0.1.1. 2021. URL:
<a href="https://github.com/RDocTaskForce/pkgcond" class="external-link">https://github.com/RDocTaskForce/pkgcond</a>.</cite>
</p>
<p>
<a id="bib-righelli2022spatialexperiment"></a><a href="#cite-righelli2022spatialexperiment">[14]</a><cite>
D. Righelli, L. M. Weber, H. L. Crowell, et al. “SpatialExperiment:
infrastructure for spatially-resolved transcriptomics data in R using
Bioconductor”. In: <em>Bioinformatics</em> 38.11 (2022), pp. -3. DOI:
<a href="https://doi.org/https://doi.org/10.1093/bioinformatics/btac299" class="external-link">https://doi.org/10.1093/bioinformatics/btac299</a>.</cite>
</p>
<p>
<a id="bib-shepherd2024biocfilecache"></a><a href="#cite-shepherd2024biocfilecache">[15]</a><cite>
L. Shepherd and M. Morgan. <em>BiocFileCache: Manage Files Across
Sessions</em>. R package version 2.13.2. 2024. DOI:
<a href="https://doi.org/10.18129/B9.bioc.BiocFileCache" class="external-link">10.18129/B9.bioc.BiocFileCache</a>.
URL:
<a href="https://bioconductor.org/packages/BiocFileCache" class="external-link">https://bioconductor.org/packages/BiocFileCache</a>.</cite>
</p>
<p>
<a id="bib-wickham2016ggplot2"></a><a href="#cite-wickham2016ggplot2">[16]</a><cite>
H. Wickham. <em>ggplot2: Elegant Graphics for Data Analysis</em>.
Springer-Verlag New York, 2016. ISBN: 978-3-319-24277-4. URL:
<a href="https://ggplot2.tidyverse.org" class="external-link">https://ggplot2.tidyverse.org</a>.</cite>
</p>
<p>
<a id="bib-wickham2023stringr"></a><a href="#cite-wickham2023stringr">[17]</a><cite>
H. Wickham. <em>stringr: Simple, Consistent Wrappers for Common String
Operations</em>. R package version 1.5.1, <a href="https://github.com/tidyverse/stringr" class="external-link uri">https://github.com/tidyverse/stringr</a>. 2023. URL:
<a href="https://stringr.tidyverse.org" class="external-link">https://stringr.tidyverse.org</a>.</cite>
</p>
<p>
<a id="bib-wickham2011testthat"></a><a href="#cite-wickham2011testthat">[18]</a><cite>
H. Wickham. “testthat: Get Started with Testing”. In: <em>The R
Journal</em> 3 (2011), pp. 5–10. URL:
<a href="https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf" class="external-link">https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf</a>.</cite>
</p>
<p>
<a id="bib-wickham2021sessioninfo"></a><a href="#cite-wickham2021sessioninfo">[19]</a><cite>
H. Wickham, W. Chang, R. Flight, et al. <em>sessioninfo: R Session
Information</em>. R package version 1.2.2, <a href="https://r-lib.github.io/sessioninfo/" class="external-link uri">https://r-lib.github.io/sessioninfo/</a>. 2021. URL:
<a href="https://github.com/r-lib/sessioninfo#readme" class="external-link">https://github.com/r-lib/sessioninfo#readme</a>.</cite>
</p>
<p>
<a id="bib-wickham2023dplyr"></a><a href="#cite-wickham2023dplyr">[20]</a><cite>
H. Wickham, R. François, L. Henry, et al. <em>dplyr: A Grammar of Data
Manipulation</em>. R package version 1.1.4, <a href="https://github.com/tidyverse/dplyr" class="external-link uri">https://github.com/tidyverse/dplyr</a>. 2023. URL:
<a href="https://dplyr.tidyverse.org" class="external-link">https://dplyr.tidyverse.org</a>.</cite>
</p>
<p>
<a id="bib-wickham2023xml2"></a><a href="#cite-wickham2023xml2">[21]</a><cite>
H. Wickham, J. Hester, and J. Ooms. <em>xml2: Parse XML</em>. R package
version 1.3.6, <a href="https://github.com/r-lib/xml2" class="external-link uri">https://github.com/r-lib/xml2</a>. 2023. URL:
<a href="https://xml2.r-lib.org/" class="external-link">https://xml2.r-lib.org/</a>.</cite>
</p>
<p>
<a id="bib-xie2024knitr"></a><a href="#cite-xie2024knitr">[22]</a><cite>
Y. Xie. <em>knitr: A General-Purpose Package for Dynamic Report
Generation in R</em>. R package version 1.48. 2024. URL:
<a href="https://yihui.org/knitr/" class="external-link">https://yihui.org/knitr/</a>.</cite>
</p>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Nicholas J. Eagles.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.9000.</p>
</div>

    </footer>
</div>





  </body>
</html>
